### PR TITLE
ci: compilation and unit test check

### PR DESCRIPTION
**Pull request by ripel2**
Linked to:
- #5 

I added a compilation check and a unit test check before mirroring to Epitech's repository
I've also added package caching to speed up our action

### CODE REVIEW

File: .github/workflows/mirror.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ on:
 
 env:
   MIRROR_URL: "git@github.com:EpitechPromo2026/G-EIP-700-NAN-7-1-eip-lucas.hauszler.git"
+  BINARY_NAME: "EngineSquared"
```
Comment (Miou-zora) [inline .github/workflows/mirror.yml:10]: What is the purpose of this line ? If it is to see if the target exists, the name should be `EngineSquared.lib` and the constant should be called `LIB_NAME` or `LIBRARY_NAME`

File: .github/workflows/mirror.yml [hunk 2]
```diff
@@ -37,11 +38,93 @@ jobs:
...
+          curl -fsSL https://xmake.io/shget.text | bash
+
+      - name: (Try to) build the project
+        run: xmake build -y
+        timeout-minutes: 2
+
+      - name: Verifies that files are present and executable
+        run: |
+          echo "FAIL_TASK=false" >> "$GITHUB_ENV"
+          SEARCH_BINARIES="${{ env.EXECUTABLES }}"
```
Comment (Miou-zora) [inline .github/workflows/mirror.yml:75]: `EXECUTABLES` ? not `BINARY_NAME` ?

File: .github/workflows/mirror.yml [hunk 3]
```diff
@@ -37,11 +38,93 @@ jobs:
...
+              echo "FAIL_TASK=true" >> "$GITHUB_ENV"
+            fi
+          done
+
+      - name: Fail if needed
+        if: env.FAIL_TASK == 'true'
+        run: exit 1
+
+  run_tests:
+    name: Run the program tests (using rule make tests_run)
```
Comment (Miou-zora) [inline .github/workflows/mirror.yml:93]: it's a copy of an old project I think because we don't use Makefile for the moment

File: xmake.lua [hunk 4]
```diff
@@ -14,4 +14,26 @@ target("EngineSquared")
 
     if is_mode("debug") then
         add_defines("DEBUG")
-    end
\ No newline at end of file
+    end
+
+for _, file in ipairs(os.files("tests/*.cpp")) do
```
Comment (Miou-zora) [inline xmake.lua:19]: Does it work for cpp files which are in folders inside the test folder?

**Review by Miou-zora — CHANGES_REQUESTED** [review]

File: xmake.lua (hunk 4 above)
Comment (ripel2) [reply xmake.lua:19]: didnt test

File: .github/workflows/mirror.yml (hunk 3 above)
Comment (ripel2) [reply .github/workflows/mirror.yml:93]: its only a typo

File: xmake.lua (hunk 4 above)
Comment (ripel2) [reply xmake.lua]: fixed

File: .github/workflows/mirror.yml (hunk 1 above)
Comment (ripel2) [reply .github/workflows/mirror.yml]: please look at the fixes I made

File: .github/workflows/mirror.yml (hunk 2 above)
Comment (ripel2) [reply .github/workflows/mirror.yml]: fixed

**Review by Miou-zora — APPROVED** [review]
LGTM

**Pull request merged**